exposed-kotlin-datetime/src/test/kotlin/org/jetbrains/exposed/sql/kotlin/datetime/DefaultsTest.kt

```diff
@@ -576,4 +576,30 @@ class DefaultsTest : DatabaseTestsBase() {
             }
         }
     }
+
+    @Test
+    fun testColumnOnUpdateCurrentTimestamp() {
+        val tester = object : Table("tester") {
+            val amount = integer("amount")
+            val created = timestamp("created").defaultExpression(CurrentTimestamp).withDefinition("ON UPDATE", CurrentTimestamp)
+        }
+
+        withTables(excludeSettings = TestDB.ALL - TestDB.ALL_MYSQL_LIKE.toSet(), tester) {
+            assertTrue { SchemaUtils.statementsRequiredToActualizeScheme(tester).isEmpty() }
```

Comment:
I would say that in `ColumnDefinitionTests` correct creation of the table is already checked, and we can trust that the correct table is created. And here we only want to check that the date was updated correctly. But it's minor, I'm also ok, if we make such check for every new type of custom column definition.

Reply:
The intention of line 588 here (and in other file tests) is not to confirm that the correct table is created, but that Exposed considers the created table column as equivalent to the retrieved metadata (by not generating unnecessary ALTER statements for the unchanged column).

It is an extra precaution though, as Exposed currently only retrieves 4 metadata: column name, whether is nullable, its default value, and whether is autoincrement. So it's unlikely, as you've seen above, that a diff will be caught; but not impossible, as with `DEFAULT ON NULL`.